tools/render-test/slang-support.cpp

```diff
@@ -119,31 +119,18 @@ static const char computeEntryPointName[] = "computeMain";
         computeTranslationUnit = translationUnit;
     }
 
-
-    Slang::List<const char*> rawGlobalTypeNames;
-    for (auto typeName : request.globalGenericTypeArguments)
-        rawGlobalTypeNames.add(typeName.getBuffer());
-    spSetGlobalGenericArgs(
-        slangRequest,
-        (int)rawGlobalTypeNames.getCount(),
-        rawGlobalTypeNames.getBuffer());
-
-    Slang::List<const char*> rawEntryPointTypeNames;
-    for (auto typeName : request.entryPointGenericTypeArguments)
-        rawEntryPointTypeNames.add(typeName.getBuffer());
-
-    const int globalExistentialTypeCount = int(request.globalExistentialTypeArguments.getCount());
-    for(int ii = 0; ii < globalExistentialTypeCount; ++ii )
+    const int globalSpecializationArgCountCount = int(request.globalSpecializationArgs.getCount());
+    for(int ii = 0; ii < globalSpecializationArgCountCount; ++ii )
     {
-        spSetTypeNameForGlobalExistentialTypeParam(slangRequest, ii, request.globalExistentialTypeArguments[ii].getBuffer());
+        spSetTypeNameForGlobalExistentialTypeParam(slangRequest, ii, request.globalSpecializationArgs[ii].getBuffer());
```

Comment:
The API function call doesn't seem to match the usage - spSetTypeNameForGlobalExistentialTypeParam, but it isn't setting  a type.

Reply:
Yes, the API name for the function was very short-sighted. At the time I had assumed that we'd want distinct APIs for type vs. value parameters, as well as existential vs. generic and global vs. entry-point.

The new COM-lite API built around `IComponentType` greatly reduces this complexity so that there is just a single linear list of specialization parameters, and the `specialize()` call takes a list of arguments that can mix types and values.

The main problem here is that we haven't updated `render-test` to use any of that new API surface area. Doing so could be a good cleanup opportunity for a future PR, but I wouldn't want to get into it here.